tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java

```diff
@@ -895,6 +898,12 @@ protected synchronized void handle(DAGAppMasterEvent event) {
     }
   }
 
+  private void handleUsedContainersOnDagFinish(DAG dag) {
+    dag.incrementDagCounter(DAGCounter.TOTAL_CONTAINERS_USED,
```

Comment:
is it actually an increment or a set? mens just from the name point of view + can this have some leftovers due to failures + retires and stuff like that?

Reply:
yeah, idempotency is always better, I'm simply creating a setCounter in DAGImpl